python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -240,12 +207,105 @@ async def setup_calsys(self, sequence_name: str) -> None:
                 distance=self.ls_select_laser_location, timeout=self.long_timeout
             )
-            await self.setup_laser(config_data["laser_mode"])
-            await self.rem.tunablelaser.cmd_startPropagating.start(
-                timeout=self.long_long_timeout
+            await self.setup_laser(
+                config_data["laser_mode"],
+                config_data["wavelength"],
+                config_data["optical_configuration"],
+            )
+            await self.laser_start_propagate()
+
+    def calculate_laser_focus_location(self, wavelength: float | None) -> float:
+        """Calculates the location of the linear stage that provides the focus
+        for the laser projector. This location is dependent on the
+        wavelength of the laser.
+
+        Parameters
+        ----------
+        wavelength : `float`
+            wavelength of the laser projector in nm
+
+        Returns
+        ------
+        location of the linear stage for the laser projector focus in mm
+
+        """
+        # TODO (DM-44772): implement the actual function
+        return 10.0
+
+    async def change_laser_wavelength(
+        self,
+        wavelength: float | None,
+        use_projector: typing.Optional[bool] = True,
+    ) -> None:
+        """Change the TunableLaser wavelength setting
+
+        Parameters
+        ----------
+        wavelength : `float`
+            wavelength of the laser in nm
+        use_projector : `bool`
+            identifies if you are using the projector while
+            changing the wavelength
+        """
+        if wavelength is not None:
+            task_wavelength = self.rem.tunablelaser.cmd_changeWavelength.set_start(
+                wavelength=wavelength, timeout=self.long_long_timeout
+            )
+
+        if use_projector:
+            task_focus = self.linearstage_laser_focus.cmd_moveAbsolute.set_start(
+                distance=self.calculate_laser_focus_location(wavelength),
+                timeout=self.long_long_timeout,
             )
+            await asyncio.gather(task_wavelength, task_focus)
 
-    async def setup_laser(self, mode: LaserDetailedState) -> None:
+        else:
+            await task_wavelength
+
+    async def change_laser_optical_configuration(
+        self, optical_configuration: LaserOpticalConfiguration
+    ) -> None:
+        """Change the output of the laser.
+
+        Parameters
+        ----------
+        optical_configuration : LaserOpticalConfiguration
+        """
+        current_configuration = (
+            await self.rem.tunablelaser.evt_opticalConfiguration.aget().configuration
+        )
+        if current_configuration != optical_configuration:
+            self.log.debug(
+                f"Changing optical configuration from {current_configuration} to {optical_configuration}"
+            )
+            if optical_configuration in list(LaserOpticalConfiguration):
```

Comment:
Since we already enforced that the `optical_configuration` is a LaserOpticalConfiguration in parameter definition, you don't actually need to perform this check here. If `optical_configuration` is not in the list when the user calls this method, it will fail.